File: 2024/01Jan/gfg/01_array-pair-sum-divisibility-problem.java
import java.util.*;

class Solution {
    public boolean canPair(int[] nums, int k) {
        // Code here
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for(int i=0;i<nums.length;i++) {
            int x = nums[i]%k;
            if(map.containsKey(x) && map.get(x)>0) {
                map.put(x, map.get(x)-1);
            }else map.put((k-x)%k, 1);
        }
        
        for(int key : map.keySet()) {
            if(map.get(key) > 0) return false;
        }
        return true;
    }
}

File: 2024/01Jan/gfg/01_array-pair-sum-divisibility-problem.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */

class Solution {
    canPair(nums,k){
        //code here
        const map = new Map();
        for(let e of nums) {
            let x = e%k;
            if(map.get(x)) map.set(x, map.get(x)-1);
            else map.set((k-x)%k, 1);
        }
        for(let [_, v] of map) if(v) return false;
        return true;
    }
}